server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/VacationRelayIntegrationTest.java

```diff
@@ -0,0 +1,192 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.google.common.base.Charsets;
+import com.jayway.awaitility.Awaitility;
+import com.jayway.awaitility.Duration;
+import com.jayway.awaitility.core.ConditionFactory;
+import com.jayway.restassured.builder.RequestSpecBuilder;
+import com.jayway.restassured.http.ContentType;
+import org.testcontainers.shaded.com.google.common.net.InetAddresses;
+
+public abstract class VacationRelayIntegrationTest {
+
+
+    @Rule
+    public final SwarmGenericContainer fakeSmtp = new SwarmGenericContainer("weave/rest-smtp-sink:latest");
+
+    private static final String DOMAIN = "mydomain.tld";
+    private static final String USER = "benwa";
+    private static final String USER_WITH_DOMAIN = USER + '@' + DOMAIN;
+    private static final String PASSWORD = "secret";
+    private static final String REASON = "Message explaining my wonderful vacations";
+
+    private static final String LOCALHOST_IP = "127.0.0.1";
+    private static final int SMTP_PORT = 1025;
+
+    private ConditionFactory calmlyAwait;
+    private GuiceJamesServer guiceJamesServer;
+    private JmapGuiceProbe jmapGuiceProbe;
+
+    private RequestSpecification jmapRequestSpecification;
+    private RequestSpecification restSmtpSinkSpec;
+
+    protected abstract void await();
+
+    protected abstract GuiceJamesServer getJmapServer();
+
+    protected abstract InMemoryDNSService getInMemoryDns();
+
+    @Before
+    public void setUp() throws Exception {
+
+        InetAddress containerIp = InetAddresses.forString(fakeSmtp.getIp());
+        getInMemoryDns()
+            .registerRecord("yopmail.com", containerIp, "yopmail.com");
+
+        guiceJamesServer = getJmapServer();
+        guiceJamesServer.start();
+
+        DataProbe dataProbe = guiceJamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(DOMAIN);
+        dataProbe.addUser(USER_WITH_DOMAIN, PASSWORD);
+        MailboxProbe mailboxProbe = guiceJamesServer.getProbe(MailboxProbeImpl.class);
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, USER_WITH_DOMAIN, DefaultMailboxes.SENT);
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, USER_WITH_DOMAIN, DefaultMailboxes.INBOX);
+        await();
+
+        jmapGuiceProbe = guiceJamesServer.getProbe(JmapGuiceProbe.class);
+        jmapRequestSpecification = new RequestSpecBuilder()
+        		.setContentType(ContentType.JSON)
+        		.setAccept(ContentType.JSON)
+        		.setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8)))
+        		.setPort(jmapGuiceProbe
+                    .getJmapPort())
+        		.build();
+
+        restSmtpSinkSpec = new RequestSpecBuilder()
+            .setContentType(ContentType.JSON)
+            .setAccept(ContentType.JSON)
+            .setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8)))
+            .setPort(80)
```

Comment:
This is not for jmap this is the port of the rest http service exposed by the docker that run rest-smtp-sink.
I think it will be hard to change it, and I don't see any good reason to do that